src/main/drivers/bus_spi_hal.c

```diff
@@ -302,45 +250,45 @@ bool spiTransfer(SPI_TypeDef *instance, uint8_t *out, const uint8_t *in, int len
 void spiSetDivisor(SPI_TypeDef *instance, uint16_t divisor)
 {
     SPIDevice device = spiDeviceByInstance(instance);
-    if (HAL_SPI_DeInit(&spiHardwareMap[device].hspi) == HAL_OK)
+    if (HAL_SPI_DeInit(&spiDevice[device].hspi) == HAL_OK)
     {
     }
 
     switch (divisor) {
     case 2:
-        spiHardwareMap[device].hspi.Init.BaudRatePrescaler = SPI_BAUDRATEPRESCALER_2;
```

Comment:
From documentation, ffs(0) == 0, ffs(1) == 1, ffs(2) == 3, ffs(4) == 3 - it seems your calculation is off by one (non-hal code has same problem)

Also you should pad the array to handle 0 and 1 case correctly. bit OR with 256 will limit max value of FFS and eliminate divisor == 0 case:
`(uint8_t []){ 0, 0, SPI_BAUDRATEPRESCALER_2, .. } [ffs(divisor | 0x100)];`